File: doc-examples/jdbc-sqlite/src/main/java/example/Person.java
package example;

import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import org.jspecify.annotations.NonNull;
import org.jspecify.annotations.Nullable;

@MappedEntity
public record Person(@Id @GeneratedValue @Nullable Long id, @NonNull String name, @NonNull Integer age) {
}

File: doc-examples/nitrite-example-kotlin/src/main/kotlin/example/Person.kt
package example

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity

// tag::person[]
@MappedEntity
class Person {
    @Id
    @GeneratedValue
    var id: String? = null

    var name: String = ""

    var age: Int = 0

    var interests: MutableList<String>? = null

    constructor()

    constructor(name: String, age: Int) {
        this.name = name
        this.age = age
    }
}
// end::person[]
